File: mysql_examples/recipes/metadata/get_enumorset_info.py
#!/usr/bin/python
# get_enumorset_info.py - wrapper to demonstrate get_enumorset_info()
# utility routine.

# Usage: get_enumorset_info.py db_name tbl_name col_name

import sys
import MySQLdb
import Cookbook
from Cookbook_Utils import *


if len (sys.argv) != 4:
  print "Usage: get_enumorset_info.py db_name tbl_name col_name"
  sys.exit (1)
db_name = sys.argv[1]
tbl_name = sys.argv[2]
col_name = sys.argv[3]

try:
  conn = Cookbook.connect ()
except MySQLdb.Error, e:
  print "Message:", e.args[1]
  print "Code:", e.args[0]
  sys.exit (1)

#@ _USE_FUNCTION_
info = get_enumorset_info (conn, db_name, tbl_name, col_name)
print "Information for " + db_name + "." + tbl_name + "." + col_name + ":"
if info == None:
  print "No information available (not an ENUM or SET column?)"
else:
  print "Name: " + info["name"]
  print "Type: " + info["type"]
  print "Legal values: " + ",".join (info["values"])
  if info["nullable"]:
    print "Nullable: yes"
  else:
    print "Nullable: no"
  if info["default"] == None:
    print "Default value: NULL"
  else:
    print "Default value: " + info["default"]
#@ _USE_FUNCTION_

conn.close ()

File: mysql_examples/recipes/metadata/get_enumorset_info.php
<?php
# get_enumorset_info.php - wrapper to demonstrate get_enumorset_info()
# utility routine.

# Usage: php get_enumorset_info.php db_name tbl_name col_name

require_once "Cookbook.php";
require_once "Cookbook_Utils.php";

if ($argc != 4)
  die ("Usage: get_enumorset_info.php db_name tbl_name col_name\n");

$conn =& Cookbook::connect ();
if (PEAR::isError ($conn))
  die ("Cannot connect to server: " . $conn->getMessage ());
$info = get_enumorset_info ($conn, $argv[1], $argv[2], $argv[3]);
printf ("Information for %s.%s.%s:\n", $argv[1], $argv[2], $argv[3]);
if (!$info)
  print ("No information available (not an ENUM or SET column?)\n");
else
{
  print ("Name: " . $info["name"] . "\n");
  print ("Type: " . $info["type"] . "\n");
  print ("Legal values: " . join (",", $info["values"]) . "\n");
  print ("Nullable: " . ($info["nullable"] ? "yes" : "no") . "\n");
  printf ("Default value: %s\n",
      ($info["default"] === NULL ? "NULL" : $info["default"]));
}

$conn->disconnect ();
?>
